src/utils.jl

```diff
@@ -134,20 +134,50 @@ end
 
 Check whether a given LPDO{MPO}  
 """
-ischoi(M::LPDO{MPO}) = (length(inds(M.X[1], "Site")) == 2 && haschoitags(M)) # hastags(M.X[1], default_purifier_tags))
+function ischoi(M::LPDO{MPO})
+  !haschoitags(M) && return false
+  for j in 1:length(M)
+    length(inds(M.X[j], "Site")) != 2 && return false
+  end
+  return true
+end
+
+function ischoi(M::MPO) 
+  !haschoitags(M) && return false
+  for j in 1:length(M)
+    length(inds(M[j], "Site")) != 4 && return false
```

Comment:
When would it not have the Choi tags? What case is this catching?

Reply:
I can't remember on the top of my head. It looks like a redundant check, but I can confirm, and it so, remove.